### PR TITLE
added createSession.js() and continueSession.js()

### PR DESCRIPTION
**Pull request by ArtskydJ**
- moved authenticate.js, isAuthenticated.js, unauthenticate.js to ./lib/
- added lib/createSession.js, lib/continueSession.js
- removed `xtend` dependency
- added `random-uuid-v4` dependency

### CODE REVIEW

**Comment by ArtskydJ** [comment]
Fixes #3.

**Comment by TehShrike** [comment]
Right now we rely on the fact that a session key exists as indicating that the session is logged in.  We would need another test to make sure that creating a session wouldn't trick the code into thinking that the session was logged in with an email address of `'null'`.

I'm still not comfortable with storing "this is a valid session" in the same key as "this session is logged in".  We should either make a new level-spaces space for noting whether or not the session is valid, or maybe better, instead of just storing the email address, store an object with a `createdAt` property and an optional `loggedInAs` property.

I don't think `continueSession` is needed - it doesn't do anything conceptually different from `isAuthenticated`.

We would need to change `unauthenticate` so that it doesn't delete the session from the db, and add a new `deleteSession` method.

**Comment by ArtskydJ** [comment]
**Note:** The [documentation for the constructor](https://github.com/coding-in-the-wild/just-login-session-state/tree/added-create-and-continue#core--sessionstatecore-db-options) needs to be updated; the options are renamed, and there are more now. Also, the link to `expire-unused-keys` is very broken.

**Comment by TehShrike** [comment]
I haven't read over the last few commits yet - do you want me to check 'em out and merge them if they look good, or do you have anything else you want to get in this pull request?

**Comment by ArtskydJ** [comment]
I'm fine with you merging this PR.
If you don't merge within the next few days, I'll probably update the documentation and push it to this branch.
If you do, I'll either make a separate PR for the docs, or commit them to master.

File: index.js [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
 
-	var expirer = new Expirer(
-		options.sessionUnauthenticatedAfterMsInactivity,
+	var authedSessionsExpirer = new Expirer(
+		opts.unauthenticateAfterMs || defaultInactivity,
+		spaces(db, 'authed-session-expiration'),
+		opts.checkIntervalMs || defaultInterval
+	)
+	var sessionsExpirer = new Expirer(
+		opts.deleteSessionAfterMs || defaultInactivity,
```
Comment (ArtskydJ) [inline index.js:24]: **Major changes:**
Renamed `opts.sessionUnauthenticatedAfterMsInactivity` -> `opts.unauthenticateAfterMs`
Added `opts.deleteSessionAfterMs`.

**Comment by ArtskydJ** [comment]
Docs have been fixed. Merge at will. :wink:

**Comment by TehShrike** [comment]
Cool!  Still needs to be published with a major version bump.

**Pull request merged**